test/e2e/autoscaler_test.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	"k8s.io/utils/ptr"
+	capi_e2e "sigs.k8s.io/cluster-api/test/e2e"
+	"sigs.k8s.io/cluster-api/test/framework"
+)
+
+var _ = Describe("When using the autoscaler with Cluster API using ClusterClass and scale to zero [vcsim] [supervisor] [PR-Blocking] [ClusterClass]", func() {
```

Comment:
This will also run with govmomi, right? As we don't want to implement it (and run it?) for govmomi, we need a a way to skip it there?

Wondering if we can just skip within the describe for govmomi

Reply:
Just saw it now. WIll the Skip below skip the entire spec?

(Can we do this earlier?)

Reply:
we can do it earlier but then we can't use `Skip` there (ginkgo limitation) and it won't be visible in the e2e test as skipped then.

Reply:
Okay, then fine as is